quail/utils.py

```diff
@@ -19,6 +24,73 @@
 logger.addHandler(handler)
 
 
+def collect_literal_inputs(request):
+    literal_inputs = [
+        request.inputs[k][0].data
+        for k in request.inputs.keys()
+        if "data_type" in vars(request.inputs[k][0]).keys()
+    ]
+    return literal_inputs
+
+
+def validate_vector(vector):
+    try:
+        vect = robjects.r(vector)
+        if robjects.r["is.vector"](vect)[0]:
+            pass
+        else:
+            raise ProcessError("Invalid type passed for vector")
+
+    except RParsingError as e:
+        raise ProcessError(
+            msg=f"{type(e).__name__}: Invalid vector format, follow R vector syntax"
+        )
+
+
+def load_ci(climdex_input, ci_name):
+    try:
+        ci = load_rdata_to_python(climdex_input, ci_name)
+        if ci.rclass[0] == "climdexInput":
+            return ci
+        else:
+            raise ProcessError(
+                msg="Input for ci-name is not a valid climdexInput object"
+            )
+
+    except RRuntimeError as e:
+        logger.error(f"cannot load {ci_name} from {climdex_input}")
+        raise ProcessError(
+            msg=f"{type(e).__name__}: Either your file is not a valid Rdata file or the climdexInput object name is not found in this rda file"
+        )
+
+
+def load_rda(file_, obj_name):
+    try:
+        return load_rdata_to_python(file_, obj_name)
+    except RRuntimeError as e:
+        err_name = re.compile(r"object \'(.*)\' not found").findall(str(e))
+        if "_" in err_name[0]:
+            raise ProcessError(
+                msg=f"{type(e).__name__}: One of the variable names passed is not an object found in the given rda files"
+            )
+        else:
+            raise ProcessError(
+                msg=f"{type(e).__name__}: There is no object named {err_name[0]} in this rda file"
+            )
+
+
+def r_valid_name(robj_name):
+    """The R function 'make.names' will change a name if it
+    is not syntactically correct and leave it if it is
+    """
+    base = get_package("base")
+    if base.make_names(robj_name)[0] != robj_name:
+        raise ProcessError(msg="Your vector name is not a valid R name")
+
+
+# Teting functions
```

Comment:
_Testing_

Reply:
Perhaps we should consider moving test specific functions into their own module.